packages/clerk-js/src/core/clerk.ts

```diff
@@ -948,7 +952,36 @@ export default class Clerk implements ClerkInterface {
     this.#componentControls?.updateProps(props);
   };
 
-  #loadInStandardBrowser = async (): Promise<void> => {
+  #redirectToSyncWithPrimary = () => {
+    const q = new URLSearchParams({
+      redirect_url: `${window.location.href}?synced=true`,
+    });
+    window.location.replace(new URL(`/v1/client/sync?${q.toString()}`, `https://${this.domain}`).toString());
+  };
+
+  #stripSyncedQueryParam = (q: URLSearchParams) => {
+    q.delete('synced');
+    const queryString = q.toString() ? `?${q.toString()}` : '';
+    window.history.replaceState({}, '', `${window.location.pathname}${queryString}`);
+  };
+
+  #shouldSyncWithMain = () => {
```

Comment:
:upside_down_face: The name of this function might be misleading. It actually performs the redirect.

I imagine "shouldSyncWithPrimary" means that I am a satellite and there's no `synced` query string parameter.

Then I can "syncWithPrimary".